sriov/sriov.go

```diff
@@ -361,180 +94,45 @@ func moveIfToNetns(ifname string, netns ns.NetNS) error {
 	return nil
 }
 
-func getDeviceNameFromPci(pciaddr string) (string, error) {
-	var devName string
-	vfDir := fmt.Sprintf("/sys/bus/pci/devices/%s/net/", pciaddr)
-	_, err := os.Lstat(vfDir)
-	if err != nil {
-		return devName, fmt.Errorf("cannot get a network device with pci address %v %q", pciaddr, err)
-	}
-	dirContents, _ := ioutil.ReadDir(vfDir)
-
-	if err != nil || len(dirContents) < 1 {
-		return devName, fmt.Errorf("failed to get network device name in %v %v", vfDir, err)
-	}
-
-	if len(dirContents) < 1 {
-		return devName, fmt.Errorf("no network device found in %v", vfDir)
-	}
+func setupVF(conf *sriovtypes.NetConf, pfName string, podifName string, cid string, netns ns.NetNS) error {
 
-	devName = dirContents[0].Name() // assuming one net device in this directory
-	return strings.TrimSpace(devName), nil
-}
-
-func setupWithVfInfo(conf *NetConf, netns ns.NetNS, cid, podifName string) error {
-	var err error
-
-	vfPciAddr := conf.DeviceInfo.PCIaddr
-	pfName := conf.DeviceInfo.Pfname
-	vfId := int(conf.DeviceInfo.Vfid)
-
-	// Get PF link with given name
 	m, err := netlink.LinkByName(pfName)
 	if err != nil {
-		return fmt.Errorf("failed to lookup master %q: %v", pfName, err)
+		return fmt.Errorf("failed to lookup master %q: %v", conf.Master, err)
 	}
 
-	// Get VF link name
-	vfLinkName, err := getDeviceNameFromPci(vfPciAddr)
+	vfLinks, err := utils.GetVFLinkNames(pfName, conf.DeviceInfo.Vfid)
```

Comment:
perhaps add a field of 'vfName' in deviceInfo structure and retrieve it inside loadConf where GetVfInfo is called and all the other VF related information are deduced there.